packages/analytics-js-integrations/src/integrations/RedditPixel/utils.js

```diff
@@ -0,0 +1,15 @@
+import { removeUndefinedAndNullAndEmptyValues } from '../../utils/commonUtils';
+const createUserIdentifier = (traits = {}) => {
+  const userIdentifier = {};
+  const { email, externalId, idfa, aaid } = traits;
+  userIdentifier.email = email;
+  userIdentifier.externalId = externalId;
+  userIdentifier.idfa = idfa;
+  userIdentifier.aaid = aaid;
```

Comment:
Is it correct to get idfa and aaid from traits? AFAIK we should get this from `context.device `

Reply:
+1

Reply:
I can see we are only using `idfa` from `context.device` [here](https://www.rudderstack.com/docs/sources/event-streams/cloud-apps/appsflyer/#:~:text=idfa-,context.device.advertisingId,-(for%20iOS)). Added that.